java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/job/creation/CreateCompactionJobsIT.java

```diff
@@ -112,6 +123,36 @@ public void shouldCompactAllFilesInSinglePartition() throws Exception {
                 });
     }
 
+    @Test
+    public void shouldSendAssignJobIdRequestToSqs() throws Exception {
+        // Given
+        tableProperties.set(COMPACTION_JOB_ID_ASSIGNMENT_COMMIT_ASYNC, "true");
+        FileReferenceFactory fileReferenceFactory = FileReferenceFactory.from(stateStore);
+        FileReference fileReference1 = fileReferenceFactory.rootFile("file1", 200L);
+        FileReference fileReference2 = fileReferenceFactory.rootFile("file2", 200L);
+        FileReference fileReference3 = fileReferenceFactory.rootFile("file3", 200L);
+        FileReference fileReference4 = fileReferenceFactory.rootFile("file4", 200L);
+        List<FileReference> files = List.of(fileReference1, fileReference2, fileReference3, fileReference4);
+        stateStore.addFiles(files);
+
+        // When
+        jobCreator().createJobs(tableProperties);
+
+        // Then
+        assertThat(stateStore.getFileReferencesWithNoJobId())
+                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
+                .containsExactlyInAnyOrderElementsOf(files);
+        CompactionJob job = receiveJobQueueMessage().getMessages().stream()
+                .map(this::readJobMessage)
+                .findFirst().orElseThrow();
+        assertThat(job.getInputFiles()).containsExactlyInAnyOrder("file1", "file2", "file3", "file4");
+        assertThat(job.getPartitionId()).isEqualTo("root");
+        CompactionJobIdAssignmentCommitRequest jobIdAssignRequest = receiveAssignJobIdQueueMessage().getMessages().stream()
+                .map(this::readAssignJobIdRequest)
+                .findFirst().orElseThrow();
```

Comment:
We can assert on the full list, for both the commit requests and the jobs. Currently it's only reading the first element of the list.